### ISSUE TITLE
[Bug] relationship field expect 'entity' parameter

### ISSUE THREAD

**Issue by mahasadhu** [Laravel-Backpack/CRUD#3216]
# Bug report

### What I did

Creating a relationship field using
```
        CRUD::field('document_category')
            ->label(trans('backpack::base.document_category'))
            ->type('relationship');
```

Just like in documentation. Without optional parameters. https://backpackforlaravel.com/docs/4.1/crud-fields#relationship
![image](https://user-images.githubusercontent.com/13914485/93657684-2e162480-fa67-11ea-962b-94137b77e2e8.png)

### What I expected to happen

Relationship field created. and I can save data normally

### What happened

When click on submit, this error shows up
SQL error `SQLSTATE[42S22]: Column not found: 1054 Unknown column 'document_category' in 'field list' (SQL: insert into 'documents' `
See https://flareapp.io/share/oPRezdmV#F66

### What I've already tried to fix it

Changing the relationship field declaration to:
```
        CRUD::field('document_category_id')
            ->label(trans('backpack::base.document_category'))
            ->type('relationship')
            ->entity('document_category');
```
So the `field` name is the foreign key column name.
And the `entity` is the relationship method declared in model.

Relationship method:
```
    public function document_category()
    {
        return $this->belongsTo('App\Models\Document_category', 'document_category_id');
    }
```

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:
### PHP VERSION:
PHP 7.4.5 (cli) (built: Apr 14 2020 16:17:34) ( ZTS Visual C++ 2017 x64 )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies

### LARAVEL VERSION:
v7.16.1@dc9cd8338d222dec2d9962553639e08c4585fa5b

### BACKPACK VERSION:
4.1.25@23fb1ed5153d5465d49d2fe1ca265a5985c78877

**Comment by mahasadhu** [comment]
And also, it looks like most of Backpack functionality regarding relationship expect the relationship method in snake_case. Previously I was using camelCase `documentCategory` as a method name in the model, and it throws this error https://flareapp.io/share/RmrMZN5z#F84
After I changed it to snake_case, this `Undefined index: relation_type` error is gone, but on save it shows SQL error as I reported above. https://github.com/Laravel-Backpack/CRUD/issues/3216#issue-704779842

This also happened when using `inline create` with camelCase method name. Backpack cannot generate the route properly, and shows this error on create page (before clicking save button) https://flareapp.io/share/W7zYQkPw#F92

Perhaps it's worth to also mention these in docs, because the error shown is not clear enough especially for early Backpack and/or Laravel users.

I hope I explain clear enough. Thank you!

**Comment by DoDSoftware** [comment]
@mahasadhu Do your models indicate that they do not use snakecase with a property such as this?


```

    /**
     * Indicates whether attributes are snake cased on arrays.
     *
     * @var bool
     */
    public static $snakeAttributes = false;

```


If your models dont have this property, could you please add it to them and re-test?

**Comment by pxpm** [comment]
@mahasadhu this issue is beeing addressed in #3199 

That error with column not found in db happens because in the current code we save BelongsTo after the item is created, and it should be done at the same time. 

About the snake cases and camel cases, we have a plan for 4.2 to remove them completely so it will be developer responsability to provide proper names for it.

**Comment by mahasadhu** [comment]
Hi @pxpm thank you for answering!

Ah I see. That was the first time I use relationship field. Previously I always use `select2` field, either with or without ajax. So I never experienced this error.

> About the snake cases and camel cases, we have a plan for 4.2 to **remove them completely** so it will be developer responsability to provide proper names for it.

I'm sorry I did not get it, what did you mean by "them" in "remove them completely" ? 
I think the solution is just to make it clear in the docs, that method name in Models should be in snake_case. Because again, the error shown is not helping new users in backpack.

Beside the docs, you can also update the generator to show additional comment.
![image](https://user-images.githubusercontent.com/13914485/94015417-68712000-fddf-11ea-8a7d-97f10434461b.png)
Maybe from this:
```
    /*
    |--------------------------------------------------------------------------
    | RELATIONS
    |--------------------------------------------------------------------------
    */
```
to this:
```
    /*
    |--------------------------------------------------------------------------
    | RELATIONS
    | Important: Methods declared have to be in snake_case.
    |--------------------------------------------------------------------------
    */
```

**Comment by pxpm** [comment]
Thanks for the detailed info. 

I could not reproduce your first error regarding the camel case:

```php
//this will only save correctly after #3199 gets merged
$this->crud->addField([
            'name' => 'superCategory',
            'type' => 'relationship',
            'ajax' => true
        ]);

//alternatively for the moment:

$this->crud->addField([
            'name' => 'column_name_in_database',
            'entity' => 'superCategory',
            'type' => 'relationship',
            'ajax' => true
        ]);

public function fetchSuperCategory() {
        return $this->fetch(\Backpack\NewsCRUD\app\Models\Category::class);
    }

public function superCategory()
    {
        return $this->belongsTo(\Backpack\NewsCRUD\app\Models\Category::class, 'select');
    }
```

When using inline_create you have to keep in mind that what is initialized is the CRUD of the element you are trying to create inline.

So if you have a `CategoryCrudController` that outputs a `category` singular name, the route will be created as `category-inline-create-save` 

So, if you setup a `category` relation in your model, everything should work just fine. but if like in my example you have a `superCategory` or `any_category` as the relation name you need to specify `inline_create => [ 'entity' => 'category']` for it to pick the proper url. 


I am going to close this as a fix for the bug is beeing addressed in #3199 and the other two might be lack of documentation. I am going to have a look at it.

If you think I am wrong, and sometimes I am, please re-open.

Wish you the best,
Pedro

**Comment by mahasadhu** [comment]
@pxpm Thank you for testing and checking.

Hmm that's strange, could it be that because I'm using fluent syntax ?

Or maybe, it's because my model class name is using snake_case instead of camelCase ?
![image](https://user-images.githubusercontent.com/13914485/94777571-3f134e00-03f6-11eb-8b19-d0e0b852231a.png)

But either way, I think for now those method name case problem can easily be solved by proper documentation, so you guys can focused on other things first. 😃
